src/lang/java/class_def.rs

```diff
@@ -0,0 +1,168 @@
+            "modifiers" => {
+                modifier = parse_modifiers(member, &*component.path, &*component.package_name)
+            }
+            "class_body" | "interface_body" | "enum_body" | "annotation_body" => {
+                parse_class_body(
+                    member,
+                    &component,
+                    &mut constructors,
+                    &mut methods,
+                    &mut fields,
+                );
+            }
+            unknown_type => log_unknown_tag(unknown_type, "class"),
+        };
+    }
+
+    // Assemble the return value from what was extracted from the tree
+    Some(ClassOrInterfaceComponent {
+        component: ContainerComponent {
+            component,
+            accessor: modifier.accessor,
+            stereotype: stereotype,
+            methods,
+            container_name: instance_name,
+            line_count: end - start + 1,
+        },
+        declaration_type,
+        annotations: modifier.annotations,
+        constructors: constructors,
+        field_components: fields,
+    })
+}
+
+/// Parses the members of a class to find members, methods, blocks, etc.
+fn parse_class_body(
+    ast: &AST,
+    component: &ComponentInfo,
+    constructors: &mut Vec<MethodComponent>,
+    methods: &mut Vec<MethodComponent>,
+    fields: &mut Vec<FieldComponent>,
+) {
+    // Traverse body
+    for member in ast.children.iter() {
+        match &*member.r#type {
+            "constructor_declaration" | "static_initializer" => {
+                constructors.push(parse_method(member, component))
+            }
+            "method_declaration" => methods.push(parse_method(member, component)),
+            "field_declaration" => fields.append(&mut parse_field(member, component)),
+            "class_declaration"
+            | "interface_declaration"
+            | "enum_declaration"
+            | "annotation_declaration" => { /* None, since these were extracted + handled elsewhere */
+            }
+            unknown => log_unknown_tag(unknown, "class body"),
+        }
+    }
+}
+
+/// Parses a single field in a class
+fn parse_field(ast: &AST, component: &ComponentInfo) -> Vec<FieldComponent> {
+    let variables: Vec<String> = ast
+        .find_all_children_by_type(&["variable_declarator"])
+        .get_or_insert(vec![])
+        .iter()
+        .map(|var_decl| var_decl.find_child_by_type(&["identifier"]))
+        .filter(|var_ident| var_ident.is_some())
+        .map(|identifier| identifier.unwrap().value.clone())
+        .collect();
+    let modifier = find_modifier(ast, &*component.path, &*component.package_name);
+    let r#type = find_type(ast);
+    let component = ComponentInfo {
+        path: component.path.clone(),
+        package_name: component.package_name.clone(),
+        instance_name: component.instance_name.clone(),
+        instance_type: InstanceType::FieldComponent,
+    };
+
+    // TODO: How to handle field_name, default_value?
```

Comment:
Is this comment outdated or still an issue?

Reply:
default_value: yes. The code exists to parse it (since it's just an assignment like in a method, as I recall), but hasn't been linked up.
field_name: I don't recall, leaving myself a note to check up on it.